File: Lab1/C#/LabTest-1(C#)/Lab1 Podviaznikov/SolveClass.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab1_Podviaznikov
{
    class SolveClass
    {
        List<int> perfectNumber = new List<int>();
        const int bound = 3000;

        public static int Sum(int naturalNumber)
        {
            int sum = 0;
            for (int i = 1; i <= naturalNumber / 2; i++)
            {
                if (naturalNumber % i == 0)
                    sum += i;
            }
            return sum;
        }

        public bool isPerfect(int number)
        {
            bool var = false;
            for (int i = 0; i < perfectNumber.Count; i++)
            {
                if (perfectNumber[i] == number)
                {
                    var = true;
                    break;
                }
            }
            return var;
        }
        public void Solver()
        {
            for (int i = 1; i <= bound; i++)
            {
                if (i == Sum(i))
                    perfectNumber.Add(i);
            }
        }
    }
}

File: Lab1/Java/LabTest-1(Java)/src/lab1/SolveClass.java
package lab1;

import java.io.Console;
import java.util.ArrayList;
import java.util.List;


public class SolveClass {

	/**
	 * @param args
	 */
	
	static List<Integer> perfectNumber = new ArrayList<Integer>();
        static final int bound = 3000; 
	public static int Sum(int naturalNumber)
	{
		int sum=0;
		for(int i=1; i<=naturalNumber/2; i++)
		{
			if(naturalNumber%i==0)
				sum+=i;
		}
		return sum;
	}
        
        public boolean isPerfect(int number)
        {
            boolean var=false;
            for(int i=0; i<perfectNumber.size();i++)
            {
                if(perfectNumber.get(i)==number)
                {
                    var=true;
                    break;
                } 
            }
           return var;
        }
	public void Solver() {
		for(int i=1; i<=bound; i++)
		{
			if(i==Sum(i))
				perfectNumber.add(i);
		}
	}
}
